### ISSUE TITLE
RSSLink not working with permalinks.posts

### ISSUE THREAD

**Issue by msridhar** [spf13/hugo#532]
I seem to have run into a bug with the combination of `.RSSLink` and use of the permalinks feature.  I have the following in my `config.toml`:

``` toml
[permalinks]
  posts = "/:year/:month/:day/:title/"
```

I'm using a minor variant of the standard hyde template, which contains this in its `layouts/chrome/head.html` file:

``` html
  <!-- RSS -->
  <link href="{{ .RSSlink }}" rel="alternate" type="application/rss+xml" title="{{ .Site.Title }}" />
```

When I look at the HTML generated for one of my posts, however, I see:

``` html
  <link href="" rel="alternate" type="application/rss+xml" title="Transitive Closure" />
```

Note that the `href` attribute has no value.  I see proper links when I use `.RSSLink` in the sidebar (though there I run into #467).  The full blog source is [here](https://github.com/msridhar/research-blog).  Let me know if I can provide further details.  And thanks so much for building hugo!

**Comment by jagipson** [comment]
I'm not sure if my issue is related to this, or not. I discovered that
```{{ if .RSSLink }}
  <link rel="alternate" type="application/rss+xml" title="{{ .Title }}" href="{{ .RSSLink }}" />
{{ end }}```
was **never** producing output.

When making a best effort attempt to resolve the issue, I noted the version I was using "Hugo Static Site Generator v0.14-DEV BuildDate: 2015-03-14T05:57:15-05:00". After pulling the latest source and recompiling, the only thing that changed was the addition of new deprecation warnings regarding template functions' names (i.e. safeUrl -> safeURL, etc).  I tried rolling back to the previous tagged release, v0.13, but Hugo would not compile because of missing (unvendored) dependencies. I recompiled master again, and I'm at "Hugo Static Site Generator v0.14-DEV BuildDate: 2015-03-22T05:45:42-05:00."

None of these troubleshooting steps have resolved the issue, and it's not obvious to me from looking at site.go where the problem is. The next thing I tried was adding to my header layout:

```
{{ if .IsPage }}
{{ ` <!-- Hugo thinks this is a page --> ` }}
{{ end }}
{{ if .IsNode }}
{{ ` <!-- Hugo thinks this is a node --> ` }}
{{ end }}
```
and reran ```hugo server --watch```

I visited several pages and nodes and the HTML comment was never added to any of the pages. Then I decided to try adding ```<h3>->{{ .Type }}</h3>``` in the HTML Body, and I'm getting the "->" but it seems that ```.Type``` is always an empty string.

**Comment by bep** [comment]
The stakeholders of this bugg: What is status?

**Comment by k4rtik** [comment]
As far as my understanding goes, this should not be an issue anymore. Feedly can successfully find the feed in all kinds of pages on my site. I am on v0.14. But then I had hacked/corrected my theme as mentioned earlier.

Perhaps @msridhar might be able to say for sure about the specific problem he described.

**Comment by msridhar** [comment]
@k4rtik What exactly is the theme fix?  With hugo 0.14 and the fix [in this comment](https://github.com/spf13/hugo/issues/532#issuecomment-63202582) I get an error that `.Site.RSSLink` is not defined.

**Comment by k4rtik** [comment]
@msridhar it is to not expose alternate link for individual blog posts or pages.

Please see: https://github.com/k4rtik/liquorice/commit/224cd5bee2eafa793742f97e474f14522470fc12 for what I did in my case. This ensures that feed is served only for those pages for which Hugo provides a RSS link (homepage, sections, category/tags pages, etc.)

**Comment by msridhar** [comment]
@k4rtik with this change will Feedly still find a feed on pages for individual posts?

**Comment by msridhar** [comment]
More specifically, I'd still like the "Subscribe" button that one can enable in Firefox to work on individual post pages.  I think that looks for a `<link>` element giving the URL of the feed.

**Comment by k4rtik** [comment]
@msridhar I am not sure how Feedly discovers a feed exactly, but on my individual posts I see the icon active though the alternate link is not there if I look up the source of the page.

Why don't you try it out once?

**Comment by msridhar** [comment]
I checked, and Firefox's RSS Subscribe button does not work on individual posts with the equivalent of k4rtik/liquorice@224cd5b enabled.  @k4rtik I went to the site linked from your GitHub profile, and I don't see the button enabled there for individual posts either (though it is enabled for the main page, as expected).  When you say that you see an active Feedly icon on your individual posts, what do you mean exactly?  Do you have a browser extension installed?

Disappointingly, I tried manually introducing a `<link>` element with `rel="home"` on my post pages, and the Firefox subscribe button doesn't work in that case either; I guess it is only looking for the incorrect `rel="alternate"` tag.  I can file a separate bug with them for that issue.  

Bottom line, though it may technically violate the specification, I think hugo should support some way for templates to reference the `RSSLink` URL on individual posts.  Having `rel="alternate"` links on individual posts is extremely common on blogs, from what I can see.  Perhaps enabling the `.Site.RSSLink` variable as suggested by @halostatue would be the best solution.  

Also, the bug title mentions the permalinks.posts feature, but I'm now unsure if that has anything to do with this issue.  If not, we can rename the issue to something more appropriate.

**Comment by k4rtik** [comment]
Yeah, I was talking about the [Feedly Mini extension](https://chrome.google.com/webstore/detail/feedly-mini/ndhinffkekpekljifjkkkkkhopnjodja). Ideally it should also NOT detect feeds on individual pages as there isn't any.

I couldn't test out myself with Firefox (don't have it installed, traveling & on a slow network connection). But yes, if it doesn't detect the feed with `<link rel="home" ...>`, letting them know is a good idea.

I cannot comment on other points you raised at this point.

**Comment by jorinvo** [comment]
Glad I found this issue. Run into the same issue. Only realized it because the RSS link was not working on sub pages.
After reading this and testing, I set the `<link>` to `rel="alternate"` for all pages to support Firefox.
Also, I link to `{{ "index.xml" | relURL }}` because I want the link to work on sub pages.
Would be nice if Hugo would take care of this though.

**Comment by bep** [comment]
This issue has been automatically marked as stale because it has not been commented on for at least four months.

The resources of the Hugo team are limited, and so we are asking for your help.

If this is a **bug** and you can still reproduce this error on the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still valuable, please open a proposal at https://discuss.gohugo.io/.

This issue will automatically be closed in four months if no further activity occurs. Thank you for all your contributions.

**Comment by bep** [comment]
**Note/Update:** This issue is marked as stale, and I may have said something earlier about "opening a thread on the discussion forum". Please don't.

If this is a **bug** and you can still reproduce this error on the latest <code>release</code> or the <code>master</code> branch, please reply with all of the information you have about it in order to keep the issue open.

If this is a **feature request**, and you feel that it is still relevant and valuable, please tell us why.

**Comment by bep** [comment]
The behaviour here is as designed.

Now `RSSLink` will only return the link for pages that potentially have a RSS feed (it may be turned off, and `RSSLink` will still create a link in that case).
